src/detect.ts

```diff
@@ -12,7 +12,7 @@ export interface DetectOptions {
 }
 
 export async function detect({ autoInstall, cwd }: DetectOptions) {
-  const result = await findUp(Object.keys(LOCKS), { cwd })
+  const result = await findUp(Object.keys(LOCKS), { cwd, stopAt: cwd })
```

Comment:
If we stop at `cwd` it then loses the meaning of having `find-up`. This will also break the usage in monorepos when you cd to a subpackage and `ni` will fail to detect the agent.